Backend/index.js

```diff
@@ -885,6 +887,103 @@ app.post("/listings/:id/complete-purchase", verifyToken, async (req, res) => {
   }
 });
 
+app.post("/listings/:id/reviews", verifyToken, async (req, res) => {
+  const { content, rating, sellerId } = req.body;
+  const { id: listingId } = req.params;
+  const reviewerId = req.user.id;
+
+  if (rating < 1 || rating > 5) {
+    return res.status(400).json({ error: "Rating must be between 1 and 5" });
+  }
+
+  try {
+    // Check if the user has purchased the item
+    const transaction = await prisma.transaction.findFirst({
+      where: {
+        listingId: parseInt(listingId),
+        buyerId: reviewerId,
+      },
+    });
+
+    if (!transaction) {
+      return res
+        .status(403)
+        .json({ error: "You must purchase the item to leave a review" });
+    }
+
+    // if the user has already reviewed this listing
+    const existingReview = await prisma.review.findFirst({
+      where: {
+        reviewerId,
+        listingId: parseInt(listingId),
+      },
+    });
+
+    if (existingReview) {
+      return res
+        .status(400)
+        .json({ error: "You have already reviewed this listing" });
+    }
+
+    const review = await prisma.review.create({
+      data: {
+        content,
+        rating,
+        reviewerId,
+        sellerId: parseInt(sellerId),
+        listingId: parseInt(listingId),
+      },
+    });
+
+    // Update seller's average rating and review count
+    const reviews = await prisma.review.findMany({
+      where: { sellerId: parseInt(sellerId) },
+    });
+
+    const reviewCount = reviews.length;
+    const averageRating =
+      reviews.reduce((acc, review) => acc + review.rating, 0) / reviewCount;
+
+    await prisma.user.update({
+      where: { id: parseInt(sellerId) },
+      data: {
+        reviewCount,
```

Comment:
do you need to update reviewCount if you don't modify it?

Reply:
when the a new review is created, it updates the review count for the user by getting the new number of reviews(length).